### ISSUE TITLE
Import from cpanel backup is totally broken. 

### ISSUE THREAD

**Issue by kupietools** [virtualmin/virtualmin-gpl#682]
First off, importing from a cpanel backup ignores any databases in the backup.

 
 Secondly, though I haven't myself played with permissions at all, somehow I have lost access privileges to my site, once I got the database tables imported manually, trying to access the site (or the phpmyadmin page I installed via your script) gives 403 errors across the board. I have now removed and reinstalled from scratch this entire package several times, and on a previous install I did get further than this (php pages loaded, but there were database connection errors) and I haven't played with the privileges in anyway, so it's tough to understand what happened if it's not a bug in the import feature. I can also still access the virtualmin dashboard remotely, as well as ping some other services on that server, so it's a problem with the import of the cpanel backup. Everything set up prior to that works. 
 
 Obviously I checked the access and error logs. There are no entries at all. Something it did during the import just completely hosed all permissions: no web pages served, no accesses logged, nothing is happened. 
 
 Safe to say: it seems safe to say Virtualmin does not have a working cpanel import feature. Or, necessary documentation for it to be working is missing. 
 
 On a personal note, as someone with 4 wordpress sites in subdomains on my site, the entire reason I chose webmin/virtualmin was because it claimed to be able to import cpanel backup files. 
 
 The ease implied by the suggestion I can import a cpanel backup is belied by the fact that this has already consumed 2 days of my life, spent troubleshooting a package I never would have installed to begin with if I had known that the cpanel import doesn't really work. 
 
 The virtualmin site should not claim that it can import cpanel backups it leaves you with something very substantially nonfunctional and missing fundamental pieces of the puzzle. That's not a "migration" feature, in my book. I really would have preferred not to spend 2 days on this. I could have migrated my entire site from my previous host manually by hand onto my server's native AMP stack in much less time than that. 
 
Even with the cautions on the site that it might take some additional configuration, and my own knowledge that there will be no guarantees of any sort with FOSS software, this experience felt distinctly user-hostile. 
 
Kindly consider either fixing it, providing working instructions including the contingencies that can go wrong and what to do about them, or removing the claim from the website that virtualmin has a cpanel backup import feature. Thanks.

**Comment by iliajie** [comment]
Hello,

We've found that cPanel imports are generally very smooth for the majority of our users. However, we've addressed a few recent issues to improve the process even further.

Could you provide some details about your Virtualmin installation, including how it was installed and the operating system and version you're using? Additionally, were there any error messages returned in Virtualmin during the cPanel backup restoration? It would also be helpful to know the cPanel version used to create the backup.

**Comment by kupietools** [comment]
Thanks for your reply... I went ahead and removed everything I'd installed, and started a complete reinstall from scratch. 

Briefly off the top of my head, it was an install onto a fresh install of Debian 12 off the amd64 official live disk image currently posted on Debian's site. I used the full  install, not net install. Virtualmin was installed first by installing webmin and then virtualmin, both from the installer shell scripts offered on the webmin and virtualmin sites... I didn't realize until afterwards you could install bother with one script. The cPanel backup restoration returned no error messages, I had no idea anything was wrong until I looked and saw no databases had been imported. For cPanel, it's whatever the most recent version Bluehost is using, although I can't tell you for sure without jumping through their hoops to log all the way in. 

Which I will do, if my current reinstall effort has the same problem. If that happens I'll come back with the full specifics.

**Comment by kupietools** [comment]
Yes, a completely fresh install failed in the exact same way. The import went flawlessly. Then, trying to turn on the SSL when saving the new virtual server, was the only hiccup: 

```
Creating SSL certificate and private key ..
.. Apache SSL website failed! : Failed to open /home/kupietzc/ssl.cert.webmintmp.1018300 : Permission denied 
```

Looking at that file, it looks like it doesn't exist. I did at some point check off an option to keep SSL certs in their website directories, but at this point I can neither find it again nor recall if it was during this or a previous reinstall that I did that. 

Then, looking at the databases, there are none, they were ignored in the import.

My cpanel version I'm migrating from is 110.0 (build 14).

Attempting to load the site in a browser now gives the default Debian apache page. (So now, on three clean reinstalls, it's given me three different things upon trying to load the site: first a database error, then a 403, and now the default Debian page. )

Since it's a wordpress site, I tried to load it by giving the domain and adding `/index.php` and it gives a file not found.

Specifying by domain, 127.0.0.1, or localhost in the browser all give the same previous two results. 

Running a configuration re-check says nothing at all is wrong. 

Software package updates says nothing needs to be updated. 

I combed the logs in the system log viewer and nothing stands out as a particular error that might explain any of this. 

The apache error log is empty. 

I then deleted the virtual servers entirely and started again. 

This time, I first set the mysql server to create new users, with all permissions, for every new system user created. 

 I created the virtual server user account beforehand, rather then letting the migration create it.

**Comment by kupietools** [comment]
Yes, a completely fresh install failed in the exact same way. The import went flawlessly. Then, trying to turn on the SSL when saving the new virtual server, was the only hiccup: 

```
Creating SSL certificate and private key ..
.. Apache SSL website failed! : Failed to open /home/kupietzc/ssl.cert.webmintmp.1018300 : Permission denied 
```

Looking at that file, it looks like it doesn't exist. I did at some point check off an option to keep SSL certs in their website directories, but at this point I can neither find it again nor recall if it was during this or a previous reinstall that I did that. 

Then, looking at the databases, there are none, they were ignored in the import.

My cpanel version I'm migrating from is 110.0 (build 14).

Attempting to load the site in a browser now gives the default Debian apache page. (So now, on three clean reinstalls, it's given me three different things upon trying to load the site: first a database error, then a 403, and now the default Debian page. )

Since it's a wordpress site, I tried to load it by giving the domain and adding `/index.php` and it gives a file not found.

Specifying by domain, 127.0.0.1, or localhost in the browser all give the same previous two results. 

Running a configuration re-check says nothing at all is wrong. 

Software package updates says nothing needs to be updated. 

I combed the logs in the system log viewer and nothing stands out as a particular error that might explain any of this. 

The apache error log is empty. 

I then deleted the virtual servers entirely and started again. 

This time, I first set the mysql server to create new users, with all permissions, for every new system user created. 

I then created a new database permissions set, allowing full permissions on all databases, for the cpanel admin username. 

 I created a system group ccalled `webmin`.
 
  I then created a system user with the same username and password as my cpanel credentials, rather then letting the migration create it. I added it to the group `mysql` on a hunch.
 
 I then added a Webmin group called `webmin-users` with all modules available and converted the system user I'd just created to a webmin user in the webmin-users group.

I then re-ran the migration. It wouldn't let me unter the username manually, complaining that it already exists, so I selected the option to let it work it out from the backup. 

It failed with an error telling me the unix user already exists and to try with a different administration username. 

I then deleted the system user I had just created, again through webmin's controls. 

I then tried the migration again, entering the admin password but telling it to work out the account from the backup. 

It failed, telling me a webmin user with the same name already existed. 

I deleted the webmin user I had just created and tried the migration again with the same options. 

It reported a successful migration. 

I went on to the server options, and enabled the SSL and MariaDB options, which had not been checked. 

It gave me the same "permission denied" error for the ssl cert but other than that indicated no error. 

Opening my domain in the browser brought up the Debian Apache page. Adding `/index.php` to the url gave a server misconfiguration error (which is an improvement on the previous 404, at least it saw a php page there.)

I clicked the Databases tab in virtualadmin and there are no databases. They were not included in the migration. 

I went to the scripts tab and tried to install phpmyadmin, thinking I could just import the databases from a direct phpmyadmin export file. Install failed with ` Fatal Error!
Failed to install script : Failed to open /home/cpaneladmin/etc/php8.2/php.ini for writing : Permission denied`

Now, the directory `/home/cpaneladmin/` was just created a moment ago by virtualadmin during the migration, it wouldn't even let me run the migration if it was set up beforehand, so if the permissions are wrong, they're wrong because virtualadmin set them wrong.

So, that's where it's at. 

In conclusion: I actually didn't post here in the issues area seeking support, especially for a FOSS package. I was just trying to report that there is an issue. 

And unfortunately, this setup attempt has now unproductively consumed over three days of my time (two and a half days longer than it's taken me in the past to accomplish the same goal manually from the command line) so I don't want to spend any more time on it. So, please don't feel obligated to respond to this with any support or troubleshooting suggestions, please just consider this thread a bug report. Thanks, again, for taking the a moment to helpfully reply. I hope something I've said here helps you to find the issue (or the documentation oversight, if that's where the problem is.)

**Comment by iliajie** [comment]
> So, please don't feel obligated to respond to this with any support or troubleshooting suggestions, please just consider this thread a bug report.

Can we have a copy of your cPanel backup? It would help a lot in reproducing the bug.

> I actually didn't post here in the issues area seeking support, especially for a FOSS package. I was just trying to report that there is an issue.

We always happy to help when it's clear what's wrong, and we always persistent in fixing bugs, which at first we need to reproduce or at least understand.

> Briefly off the top of my head, it was an install onto a fresh install of Debian 12 off the amd64 official live disk image currently posted on Debian's site. I used the full install, not net install. 

You should use minimal install, as full install may pre-install quite a lot of other software.

> I created a system group ccalled webmin.

No need for doing any of this!

> Virtualmin was installed first by installing webmin and then virtualmin, both from the installer shell scripts offered on the webmin and virtualmin sites

You don't need to do anything other than:

```
wget https://software.virtualmin.com/gpl/scripts/virtualmin-install.sh
sh virtualmin-install.sh
```


> And unfortunately, this setup attempt has now unproductively consumed over three days of my time

Apologies for the inconvenience. Although this procedure should take no more than 30-45 minutes of time. I'm not sure why it took 3 days.

**Comment by shoulders** [comment]
this should be closed because:
- end user support
- I just imported from cpanel (with some minor isse whihc I am going to report)
- now 2023
